expo/tests/scheduler/test_add_event.py

```diff
@@ -1,100 +1,331 @@
-        '''Should get 404 if no valid act ID'''
-        request = self.factory.get('/scheduler/create/GenericEvent/-1')
-        request.user = self.profile_factory.create().user_object
-        response = add_event(request, -1)
-
-    @nt.raises(PermissionDenied)
-    def test_add_event_no_permission(self):
-        '''add event attempt should fail because user is not a Schedule Maven
-        '''
-        profile = self.profile_factory.create()
-        request = self.factory.get('/scheduler/create/GenericEvent/%d' %
-                                   self.eventitem.pk)
-        request.user = profile.user_object
-        response = add_event(request, self.eventitem.pk, "GenericEvent")
-
-""""
-COMMENTED OUT 9/22/2015
-Test failing, couldn't fix
-If still commented out after 10/22, kill this test and we'll try again
-
-
-    def test_add_event_submit_succeed(self):
-        '''add event post succeeds, user has proper privileges
-        '''
-        profile = self.profile_factory.create()
-        form_post = self.get_add_session_form()
-        request = self.factory.post('/scheduler/create/GenericEvent/%d' %
-                                    self.eventitem.pk,
-                                    form_post)
-        request.user = profile.user_object
-        request.session = {'cms_admin_site':1}
-        functions.grant_privilege(profile, 'Scheduling Mavens')
-        response = add_event(request, self.eventitem.pk, "GenericEvent")
-
-        self.assertEqual(response.status_code, 200)
-
-"""
-"""
-COMMENTED OUT 9/22/2015
-Test failing, couldn't fix
-If still commented out after 10/22, kill this test and we'll try again
-
-    def test_add_event_get_succeed(self):
-        '''add event get succeeds, user has proper privileges
-        '''
-        profile = self.profile_factory.create()
-        request = self.factory.get('/scheduler/create/GenericEvent/%d' %
-                                   self.eventitem.pk)
-        request.user = profile.user_object
-        functions.grant_privilege(profile, 'Scheduling Mavens')
-        request.session = {'cms_admin_site':1}
-        response = add_event(request, self.eventitem.pk, "GenericEvent")
-        self.assertEqual(response.status_code, 200)
-        self.assertTrue(self.eventitem.title in response.content)
-        self.assertTrue(self.eventitem.description in response.content)
-"""
+        self.user = ProfileFactory.create().user_object
+        self.privileged_profile = ProfileFactory()
+        self.privileged_user = self.privileged_profile.user_object
+        grant_privilege(self.privileged_user, 'Scheduling Mavens')
+        self.eventitem = GenericEventFactory()
+
+    def test_no_login_gives_error(self):
+        url = reverse(self.view_name,
+                      urlconf="scheduler.urls",
+                      args=["GenericEvent", self.eventitem.eventitem_id])
+        response = self.client.get(url)
+        nt.assert_equal(response.status_code, 302)
```

Comment:
Where does the redirect take us? Please use a follow=True and test this to its final destination